pkg/plugins/handlers.go

```diff
@@ -17,32 +17,74 @@ import (
 type PluginsHandler struct {
 	Client             *http.Client
 	PluginsEndpointMap map[string]string
+	PublicDir          string
 }
 
-func NewPluginsHandler(client *http.Client, token string, pluginsEndpointMap map[string]string) *PluginsHandler {
+func NewPluginsHandler(client *http.Client, pluginsEndpointMap map[string]string, publicDir string) *PluginsHandler {
 	return &PluginsHandler{
 		Client:             client,
 		PluginsEndpointMap: pluginsEndpointMap,
+		PublicDir:          publicDir,
 	}
 }
 
-func (p *PluginsHandler) HandlePlugins(w http.ResponseWriter, r *http.Request) {
+func (p *PluginsHandler) HandleLocales(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Method unsupported, the only supported methods is GET"})
+		return
+	}
+
+	query := r.URL.Query()
+	lang := query.Get("lng")
+	// In case of console app or static plugins, the localization namespace should contain name of the json file that contains the localized strings,
+	// since various console packages have different names for the localization file.
+	// eg. namespace query for the local-storage-operator-plugin locales - '&ns=lso-plugin'
+	// In case of the dynamic plugin the localization namespace should contain name of the plugin prefixed with 'plugin__' prefix.
+	// No file extension is needed since for the dynamic plugins we will fetch file from the plugin pod, based on the namespace.
+	// eg. 'plugin__helm' will fetch `/locales/{lang}/plugin__helm.json`
```

Comment:
I think this is the important bit, so I'd trim the comment to just this:

```suggestion
	// In case of the dynamic plugins, the namespace should contain name of the plugin prefixed with 'plugin__' prefix.
	// eg. 'plugin__helm' will fetch `locales/{lang}/plugin__helm.json` from the plugin service
```